adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,47 @@
+= Rename Reactive extensions
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+With the advent of Virtual Threads in Java 19+, the problem becomes larger since in the minds of users, Reactive and Virtual Threads
+are two different worlds.
+
+The same problem applies to other extensions as well:
+* Reactive Messaging
+* Reactive Routes
+* REST Client reactive
+
+The common theme here is that these extensions can work equally well with reactive and blocking code, but their name confuses users.
+
+We have tried to clarify the situation with blog posts, talks and on the various communication channels, but there seems to be
+no escaping the fact that "reactive" in the name primes people in a non-desirable way.
+
+
+== Considered options
+
+We can continue with the current situation, but that feels less than ideal given the recurrence of the problem.
+
+== Decision
+
+* Rename all `quarkus-resteasy-reactive*` maven modules to `quarkus-rest*`
+* Rename all `quarkus-rest-client-reactive*` maven modules to ???
+* Rename all `quarkus-smallrye-reactive-messaging-` maven modules to `quarkus-reactive-messaging-`
+* Rename all `quarkus-reactive-routes-` maven modules to `quarkus-http-routes-`
+* No Java code changes should be made - all code should continue to reside in the same packages packages
+* Introduce Maven relocations to allow for users not updating their poms
```

Comment:
could making new extensions that just include the above be a better option?

Reply:
I don't see how this helps vs relocation.

Reply:
it avoids breakage.

Reply:
If we have relocation for `quarkus-resteasy-reactive` to `quarkus-rest` (and others), why would things break?

It's the same thing we've done in the past (Reactive Routes comes to mind)

Reply:
I agree it proboably don't have an impact - if zero changes done in the code/config/package names.

The advantage a new extension with dependency on the old is ability to do some changes/additions - but if we deem that is not useful or needed then for sure all good. 

Thats just what I would then mention in the ADR as an option that was disregarded because not seen as being needed.

Reply:
👍🏼